shared/ssz/decode.go

```diff
@@ -277,26 +277,27 @@ func makeStructDecoder(typ reflect.Type) (decoder, error) {
 	return decoder, nil
 }
 
-// Notice: Currently we don't support nil pointer:
-// - Input for encoding must not contain nil pointer
-// - Output for decoding will never contain nil pointer
-// (Not to be confused with empty slice. Empty slice is supported)
 func makePtrDecoder(typ reflect.Type) (decoder, error) {
 	elemType := typ.Elem()
 	elemSSZUtils, err := cachedSSZUtilsNoAcquireLock(elemType)
 	if err != nil {
 		return nil, err
 	}
+
+	// TODO(1461): The encoding of nil pointer isn't defined in the spec.
+	// After considered the use case in Prysm, we've decided that:
+	// - We assume we will only encode/decode pointer of array, slice or struct.
+	// - The encoding for nil pointer shall be 0x00000000.
+
 	decoder := func(r io.Reader, val reflect.Value) (uint32, error) {
-		newVal := val
-		if val.IsNil() {
-			newVal = reflect.New(elemType)
-		}
+		newVal := reflect.New(elemType)
 		elemDecodeSize, err := elemSSZUtils.decoder(r, newVal.Elem())
```

Comment:
This would also be valid without performing an additional method call

```
elemDecodeSize, err := elemSSZUtils.decoder(r, elemType)
```

Reply:
It doesn't seem to work...`elemType` is `reflect.Type`, but `newVal.Elem()` is `reflect.Value`.

Reply:
Ah , thanks. I thought it would be the same type but turns out its not